components/measurement/nettests/WebConnectivity.js

```diff
@@ -271,18 +272,45 @@ QueryContainer.propTypes = {
   query: PropTypes.object
 }
 
+const validateMeasurement = (measurement) => {
```

Comment:
This validation function can either be evolved into a generic one to run before deciding to render a specific component from `measurement/nettests/*` or similar local methods across all other measurements. Right now it makes sure the component works with an object that has all the expected keys, even if they are absent in API responses.

Reply:
Maybe I should add this in the file itself.